internal/nodes/dequeue_ingestion.go

```diff
@@ -160,10 +160,36 @@ func DequeueIngestion(key string, fullsync bool) {
 
 	// L4Rule CRD processing.
 	if objType == lib.L4Rule {
+
 		svcNames, found := schema.GetParentServices(name, namespace, key)
-		if found && utils.CheckIfNamespaceAccepted(namespace) {
-			for _, svcNSNameKey := range svcNames {
-				handleL4Service(utils.L4LBService+"/"+svcNSNameKey, fullsync)
+		if !found {
+			utils.AviLog.Debugf("key: %s, msg: no service found with L4Rule annotation", key)
+			return
+		}
+
+		if !utils.CheckIfNamespaceAccepted(namespace) {
+			utils.AviLog.Debugf("key: %s, msg: namespace of l4rule is not in accepted state", key)
+			return
+		}
+
+		for _, svcNSNameKey := range svcNames {
+
+			// L4 Service
+			handleL4Service(utils.L4LBService+"/"+svcNSNameKey, fullsync)
```

Comment:
I think this should be invoked only when it is not a shared vip svc.

Reply:
I have addressed the comment.